File: backend/PhoneHub.Core/Models/SilentModeSettings.cs
namespace PhoneHub.Core.Models;

/// <summary>
/// Gizli Mod Ayarları
/// </summary>
public class SilentModeSettings
{
    public Guid Id { get; set; }
    public Guid DeviceId { get; set; }
    
    /// <summary>
    /// Gizli mod etkinleştirildi mi
    /// </summary>
    public bool IsEnabled { get; set; }

    /// <summary>
    /// Ekranı kapalı tutma
    /// </summary>
    public bool KeepDisplayOff { get; set; }

    /// <summary>
    /// Ses sistemini kapat
    /// </summary>
    public bool MuteAudio { get; set; }

    /// <summary>
    /// Titreşimi devre dışı bırak
    /// </summary>
    public bool DisableVibration { get; set; }

    /// <summary>
    /// Bildirim LED'ini kapat
    /// </summary>
    public bool DisableNotificationLED { get; set; }

    /// <summary>
    /// Sistem bildirimlerini gizle
    /// </summary>
    public bool HideSystemNotifications { get; set; }

    /// <summary>
    /// Logları kaydetme (gizli işlemler kayıt altına alınmaz)
    /// </summary>
    public bool DisableLogs { get; set; }

    /// <summary>
    /// Gizli mode PIN koruması
    /// </summary>
    public string? PIN { get; set; }

    /// <summary>
    /// PIN hash'i (salt ile birlikte)
    /// </summary>
    public string? PINHash { get; set; }

    /// <summary>
    /// Gizli işlem geçmişini sil
    /// </summary>
    public bool ClearHistoryAfterSession { get; set; }

    /// <summary>
    /// Etkinleştirilme tarihi
    /// </summary>
    public DateTime EnabledAt { get; set; }

    /// <summary>
    /// Son güncellenme
    /// </summary>
    public DateTime UpdatedAt { get; set; }

    public Device? Device { get; set; }
}

File: mobile/PhoneHub/app/src/main/java/com/phonehub/android/models/SilentModeSettings.kt
package com.phonehub.android.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.util.*

/**
 * Gizli Mod Ayarları Modeli
 */
@Parcelize
data class SilentModeSettings(
    val id: String = UUID.randomUUID().toString(),
    val deviceId: String = "",
    val isEnabled: Boolean = false,
    val keepDisplayOff: Boolean = true,
    val muteAudio: Boolean = true,
    val disableVibration: Boolean = true,
    val disableNotificationLED: Boolean = true,
    val hideSystemNotifications: Boolean = true,
    val disableLogs: Boolean = true,
    val clearHistoryAfterSession: Boolean = true,
    val pinHash: String? = null,
    val enabledAt: Long = 0,
    val updatedAt: Long = 0
) : Parcelable
